src/lib.rs

```diff
@@ -222,7 +223,7 @@ fn match_fields(
         .map(|field| {
             let label = field.label.clone();
             let ident = format_ident!("{}", &field.label);
-            let index = field.index + offset;
+            let index = field.index.expect("index must be set for fields that are not skipped") + offset;
```

Comment:
Maybe document that `parse` guarantees that this is always `Some`  and thus will not panic.

This could potentially be made a stronger guarantee by having the index be stored in the `Skip` enum.

Reply:
I’ve added a comment.  Enforcing it through the type system would be a bigger change.